src/cli.ts

```diff
@@ -14,19 +14,35 @@ export interface CliExitData {
 }
 
 export interface ICli {
-    execute(cmd: string, opts?: ExecOptions): Promise<CliExitData>;
+    execute(cmd: CliCommand, opts?: SpawnOptions): Promise<CliExitData>;
 }
 
 export interface TknChannel {
     print(text: string): void;
     show(): void;
 }
 
+export interface CliCommand {
+    cliCommand: string;
+    cliArguments: string[];
+}
+
+export function createCliCommand(cliCommand: string, ...cliArguments: string[]): CliCommand {
+    if (!cliArguments) {
+        cliArguments = [];
+    }
+    return { cliCommand, cliArguments };
+}
+
+export function cliCommandToString(command: CliCommand): string {
+    return `${command.cliCommand} ${command.cliArguments.join(' ')}`;
+}
+
 export class Cli implements ICli {
     private static instance: Cli;
     private tknChannel: TknChannel = new TknChannelImpl();
 
-    private constructor() {}
+    private constructor() { }
```

Comment:
if the contstructor is empty I think it should be removed

Reply:
I think that 'Java style' singleton pattern, we need to refactor that singleton in more TS/JS way.
But agree to remove that

Reply:
And I even not sure why we need `ICli` interface